### DISCUSSION TITLE
Issue with mapping a warped image into a reference space with buildtemplateparallel.sh - WarpImageMultiTransform

### DISCUSSION THREAD

**Discussion by DustiRog** [ANTsX/ANTs#1509] [Q&A]
I'm having some strange behavior when trying to map a warped image into a reference space using the inversion of an affine transform.  What's happening is the image is off center so only the bottom right corner is present:

![image](https://user-images.githubusercontent.com/111895447/223636327-c6f90d27-d470-440b-b512-90ac99b52d92.png)

The original warped image looks fine:
![image](https://user-images.githubusercontent.com/111895447/223636885-4086a9f3-1086-4a4a-9ad4-d5dfc697bf9f.png)

As well as the image for the reference space:
![image](https://user-images.githubusercontent.com/111895447/223636623-88b2cff8-0f27-4bb7-a322-8eb1d13a52cc.png)

So I'm thinking the affine transformation txt file is the issue? But am not too familiar with it:

#Insight Transform File V1.0
#Transform 0
Transform: MatrixOffsetTransformBase_double_3_3
Parameters: 0.8680842247191992 -0.0024614585527550324 -0.03639305305885 -0.0021814667712567364 0.8536843868982914 0.15138631811676873 0.03651728762988701 -0.20307767161779464 0.6782687858457199 -118.91556333353063 -130.68804104204435 58.4924303108248
FixedParameters: -0.46588847698700003 18.922574229943738 4.746380733313081

Input command:
${ANTSPATH}/WarpImageMultiTransform 3 D1_templatewarp.nii.gz D1_templatewarp.nii.gz -i  D1_templateAffine.txt -R D1_template.nii.gz

Does any of this look familiar to anyone? Could definitely use the help.

**Comment by ntustison** [comment]
A couple things---this really isn't enough information to diagnose the problem.   Ideally, we'd like to see the command call, simplified as much as possible, with perhaps sample data so we can reproduce the issue on our end.  It looks like you want the inverse of the affine transform but you're coupling it with twice the warp field and I don't know of any situation where that makes sense.  Second, buildtemplateparallel.sh and WarpImageMultiTransform have been deprecated for years by antsMultivariateTemplateConstruction.sh and antsApplyTransforms.  Perhaps it would also help if you explained the context for using such outdated ANTs commands.

**Reply by DustiRog** [reply]
Hey Nick,

Sorry oversimplified what's happening! So big picture I'm trying to extract a skeletonized grey matter map to project NDI and ODI onto.  To do this I am using the pipeline from Gray-matter Based Spatial Statistics (https://github.com/arash-n/GBSS).  At the very end of the first script (https://github.com/arash-n/GBSS/blob/master/NODDI/gbss_1_reg.sh) it uses build templateparallel.sh to create a study specific pseudo T1 template from the created individual pseudo T1 templates.

`###############################################################`
`### PART 1.4.a: Creating Template/ Estimating Warp Fields  ####`
`###############################################################`
`mkdir ${out_dir}/FA/D1`
`cp ${out_dir}/FA/FA/*_pseudoT1.nii.gz ${out_dir}/FA/D1/`
`cd ${out_dir}/FA/D1`

`if [ ${#method} == 1 ]`
`then`

`echo "OPTION 1: Creating Template Image"`

`buildtemplateparallel.sh  -d 3 -j 1 -o D1_ -n 0 -s MI -i 4 -m 30x50x20 -t GR -z MNI152_T1_1mm.nii.gz
  *_pseudoT1.nii.gz`

`fi`

Individual pseudo T1 templates:
[BL006_pseudoT1.nii.gz](https://github.com/ANTsX/ANTs/files/10923053/BL006_pseudoT1.nii.gz)
[BL008_pseudoT1.nii.gz](https://github.com/ANTsX/ANTs/files/10923055/BL008_pseudoT1.nii.gz)

Reference template (MNI152_T1_1mm.nii.gz renamed to match pipeline):
[D1_template.nii.gz](https://github.com/ANTsX/ANTs/files/10923608/D1_template.nii.gz)

So, looking at the antsMultivariateTemplateConstruction.sh process (line 393), it has the same process as buildtemplateparallel.sh (line 413) where it uses the WarpImageMultiTransform that seems to be where the weird transform is happening.  This makes me think the deprecated script is not the root of the issue.  

To recreate run the following code in a folder with the attachd files files:

`buildtemplateparallel.sh  -d 3 -j 1 -o D1_ -n 0 -s MI -i 4 -m 30x50x20 -t GR -z D1_template.nii.gz  *_pseudoT1.nii.gz`

Images start to look off when the step at line 413 of the buildtemplateparallel.sh is run :
`${ANTSPATH}/WarpImageMultiTransform 3 D1_templatewarp.nii.gz D1_templatewarp.nii.gz -i  D1_templateAffine.txt -R D1_template.nii.gz`

*The updated template "D1_templatewarp.nii.gz" is too big to attach so I couldn't setup this step directly.

I can try to recreate it using antsMultivariateTemplateConstruction.sh this afternoon.

Thank you,

Dustin

**Comment by ntustison** [comment]
Okay, now it makes a bit more sense as to what you're trying to do.  I would recommend that users *not* try to debug the internals of a script but rather, if the output doesn't match expectations, describe that overall problem first and we can provide guidance as to the next steps.  In your case, one of the initial problems I see is that your reference MNI template has the skull intact whereas your pseudo images do not.  That input discrepancy is going to cause issues even before you start trying to debug specific steps of the script.

**Reply by DustiRog** [reply]
You're a wizard Harry! The skull stripped template fixed it.

Thank you so much!

**Comment by ntustison** [comment]
Great.  Glad it worked.
